salt/tests/unit/modules/files/test_metalk8s_kubernetes_utils.yaml

```diff
@@ -0,0 +1,105 @@
+get_kubeconfig:
+  # 1. get kubeconfig from kwargs options only
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf", "context": "my-kwargs-mycontext"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 2. get kubeconfig from api-server pillars only
+  - api_server_pillar: {
+      "kubeconfig": "my-pillar-kubeconfig.conf",
+      "context": "my-pillar-context"
+    }
+    result: ['my-pillar-kubeconfig.conf', 'my-pillar-context']
+  # 3. get kubeconfig from salt config options only
+  - config_options: {
+      "kubernetes.kubeconfig": "my-config-option-kubeconfig.conf",
+      "kubernetes.context": "my-config-option-mycontext"
+    }
+    result: ['my-config-option-kubeconfig.conf', 'my-config-option-mycontext']
+
+  # 4. get kubeconfig partly from kwargs and salt config options
+  - kwargs: {"context": "my-kwargs-mycontext"}
+    config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    result: ['my-config-option-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 5. get kubeconfig partly from kwargs and api-server pillars
+  - kwargs: {"context": "my-kwargs-mycontext"}
+    api_server_pillar: {"kubeconfig": "my-kwargs-kubeconfig.conf"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 6. get kubeconfig partly from salt config options and context result is None
+  - config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    result: ['my-config-option-kubeconfig.conf', null]
+  # 7. get kubeconfig partly from api-server pillar and context result is None
+  - api_server_pillar: {"kubeconfig": "my-pillar-kubeconfig.conf"}
+    result: ['my-pillar-kubeconfig.conf', null]
+  # 8. get kubeconfig partly from kwargs and context result is None
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf"}
+    result: ['my-kwargs-kubeconfig.conf', null]
+  # 9. test that kwargs overrides pillar and salt config option
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf", "context": "my-kwargs-mycontext"}
+    api_server_pillar: {
+      "kubeconfig": "my-pillar-kubeconfig.conf",
+      "context": "my-pillar-context"
+    }
+    config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 10. test that kubeconfig is defined in the lowest level and context is being overridden
+  - api_server_pillar: {"context": "my-pillar-context"}
+    config_options: {
+      "kubernetes.kubeconfig": "my-config-option-kubeconfig.conf",
+      "kubernetes.context": "my-config-option-mycontext"
+    }
+    result: ['my-config-option-kubeconfig.conf', 'my-pillar-context']
+
+read_and_render_yaml_file:
+  # 1. read and render a valid yaml file
+  - source: |-
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: kube-apiserver
+    result:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: "kube-apiserver"
+  #. 2. Raise error when no source file is specified
```

Comment:
The title is wrong it's not "no source file specified" but source file does not exists